### PR TITLE
Fix CI Metrics job

### PR DESCRIPTION
**Pull request by WunderBart**
### Changes proposed in this Pull Request:

- Fix the Metrics CI job setup (see pdV5qK-zn-p2).
- Update the base commit SHA to one that is compatible with WP 6.6.
- Make the related artifacts uploadable via the generic handler, so no dedicated "Archive metrics results" step is necessary.

### How to test the changes in this Pull Request:

The Metrics job should run in CI **only against trunk** and pass. See https://github.com/woocommerce/woocommerce/pull/50214#issuecomment-2269144079 for further info.

### CODE REVIEW

File: plugins/woocommerce/package.json [hunk 1]
```diff
@@ -529,8 +529,12 @@
 						".wp-env.json"
 					],
 					"events": [
-						"disabled"
-					]
+						"pull_request"
```
Comment (WunderBart) [inline plugins/woocommerce/package.json]: @adimoldovan, this is only for testing. Let's switch to trunk once it's reviewed! (pdV5qK-zn-p2#comment-968)

Comment (adimoldovan) [reply plugins/woocommerce/package.json]: My understanding is that we're ok with keeping it on push to trunk only (pdV5qK-zn-p2#comment-968). So the right event would be `push`.

Comment (WunderBart) [reply plugins/woocommerce/package.json]: I remembered we discussed that right after submitting the comment, so I edited it. 😄 Didn't make it in time before you responded, though! Anyway, as mentioned above, let's switch to push before this is merged.

**Comment by WunderBart** [comment]
All the checks have passed for https://github.com/woocommerce/woocommerce/pull/50214/commits/3f97edf4e3a449483442f5177d746bbb18512fe0. Switching the event trigger to `push`! cc @adimoldovan 

_edit:_ On second thought, I'll run the base branch comparison here as well to be sure it doesn't break against trunk. We'll also need to update the base commit SHA as the current one is from December 2023. Will handle that early next week!

**Comment by WunderBart** [comment]
With https://github.com/woocommerce/woocommerce/pull/50214/commits/3c541c7c52bfe7631ec16ab1d758c8a210f3bd57:
<details>
<summary>Metrics job for the base branch comparison is passing</summary>


- Link: https://github.com/woocommerce/woocommerce/actions/runs/10249668786/job/28353623332?pr=50214

- Screenshot:
   
   github com_woocommerce_woocommerce_actions_runs_10249668786_job_28353623332(Desktop) (1)

</details>

<details>
<summary>Measurement files are produced and uploaded as expected</summary>

- Uploaded archive: [core-metrics-report__13.zip](https://github.com/user-attachments/files/16508495/core-metrics-report__13.zip)

- Screenshot:
   
   github com_woocommerce_woocommerce_actions_runs_10249668786_job_28353623332(Desktop) (2)

</details>

@adimoldovan, I will now make the Metrics check run against `trunk` only, which should make this PR mergeable. After merging, we'll need to monitor if the numbers on https://www.codevitals.run/project/woo are getting updated once the first `trunk` job passes. 🤞

**Review by adimoldovan — APPROVED** [review]
LGTM.
Thanks for fixing this @WunderBart!

**Pull request merged**